---
layout: page
title: Curriculum vitae
permalink: /cv/
---

**Current**

Postdoctoral Research Fellow, Harvard T.H. Chan School of Public Health

**Education**

* Ph.D. in Bioinformatics, UCLA, 2018
* B.S. in Computer Science, UCLA, 2013

**Publication**

* **Shi H.**, Mancuso N., Spendlove S., Pasaniuc B.
[Local genetic correlation gives insights into the shared genetic architecture of complex traits](http://biorxiv.org/content/early/2016/12/08/092668){:target="_blank"}

* Mancuso N., **Shi H.**, Goddard P., Kichaev G., Gusev A., Pasaniuc B.
[Integrating gene expression with summary association statistics to identify
susceptibility genes for 30 complex traits.](http://biorxiv.org/content/early/2016/09/01/072967){:target="_blank"}

* **Shi, H.**, Kichaev, G., Pasaniuc B.
[Contrasting the genetic architecture of 30 complex traits from summary association data.](http://www.cell.com/ajhg/abstract/S0002-9297(16)30148-3){:target="_blank"} (American Journal of Human Genetics, 2016)

* Gusev A., Ko Arthur., **Shi H.**, Bhatia G., Price A., Pajukanta P., Pasaniuc B., et al.
[Integrative approaches for large-scale transcriptome-wide association studies.](http://www.nature.com/ng/journal/v48/n3/full/ng.3506.html){:target="_blank"} (Nature Genetics, 2016)

* Gusev A., **Shi H.**, Kichaev G., Price A., Pasaniuc B., et al.
[Genomic functional atlas of prostate cancer heritability in European and African Americans reveals extensive tissue-specific regulation.](http://www.nature.com/ncomms/2016/160407/ncomms10979/full/ncomms10979.html){:target="_blank"} (Nature Communication, 2016)

* **Shi, H.**, Pasaniuc, B., Lange, K.
[A multivariate Bernoulli model to predict DNaseI hypersensitivity status from haplotype data.](http://bioinformatics.oxfordjournals.org/content/31/21/3514){:target="_blank"} (Bioinformatics, 2015)

* Pasaniuc B., Zaitlen N., **Shi H.**, Bhatia G., Gusev A., Pickrell
J., Hirschhorn J., Strachan D.P., Patterson N., Price A.L.
[Fast and accurate imputation of summary statistics enhances evidence of functional enrichment.](http://bioinformatics.oxfordjournals.org/content/early/2014/07/18/bioinformatics.btu416){:target="_blank"} (Bioinformatics, 2014)

* Orozco L.D., Morselli M., Rubbi L., Guo W., Go J., **Shi H.**, Lopez
D., Furlotte N.A., Bennett B.J., Farber C.R., Ghazalpour A., Zhang
M.Q., Bahous R., Rozen R., Lusis A.J., Pellegrini M.
[Epigenome-wide association of liver methylation patterns and complex metabolic
traits in mice.](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4454894/){:target="_blank"} (Cell Metabolism, 2015)

**Software**

* [HESS: Heritability Estimation from Summary Statistics](http://github.com/huwenboshi/hess){:target="_blank"}
* [ImpG-Summary: Gaussian imputation of summary statistics](http://github.com/huwenboshi/impg){:target="_blank"}
* I am also involved in [TWAS](http://bogdan.bioinformatics.ucla.edu/software/twas/){:target="_blank"}.

**Skills**

* C, C++, Java, CUDA, Matlab, Python, R, Unix shell script, Graphics Processing Unit, Computing Cluster

**Courses**

* Applied Probability, Bayesian Statistics, Theoretical Statistics, Statistical
Methods in Computational Biology, Multivariate Analysis, Machine
Learning and Pattern Recognition
* Linear Programming, Convex Optimization, Optimization Methods for
Large-Scale Systems, Monte Carlo Methods for Optimization, Optimization
Methods in Biology, Matrix Algebra and Optimization
